File: springboot-java/src/main/java/hello/GreetingService.java
package hello;

import org.springframework.stereotype.Service;

@Service
class GreetingService {
    final NameFormatter formatter;

    public GreetingService(NameFormatter formatter) {
        this.formatter = formatter;
    }

    public String greet(String name) {
        return "Hello, " + formatter.format(name) + "!";
    }
}

File: springboot-kotlin/src/main/kotlin/hello/GreetingService.kt
package hello;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

@Service
class GreetingService {
    @Autowired
    lateinit var formatter: NameFormatter

    fun greet(name: String) = "Hello, ${formatter.format(name)}!"
}
